sse_client.js

```diff
@@ -178,49 +180,53 @@ module.exports = function(RED) {
                     url = node.url;
                     node.warn('Warning: msg properties can not override set node properties. Using set node properties.');
                 } else {
-                    node.status({fill: "red", shape: "dot", text: "no url"});
+                    node.status({ fill: "red", shape: "dot", text: "no url" });
                     return;
                 }
-                
+
                 if (isTemplatedUrl) {
                     url = mustache.render(url, msg);
                 }
-                    
+
                 // Start a new stream (i.e. send a new http get)
                 node.client = new EventSource(url, options);
 
-                node.client.onopen = function() {
-                    node.status({fill: "green", shape: "dot", text: "connected"});
+                node.client.onopen = function () {
+                    setTimeout(() => {
+                        node.status({ fill: "green", shape: "dot", text: "connected" });
+                    }, 1000); // 08/04/2022 Supergiovane: Allow the "reconnecting" status to be shown, in case of reconnection by timeout.
+
                 }
 
-                node.client.onerror = function(err) {
-                    node.status({fill: "red", shape: "dot", text: `Error: ${err.message}`});
+                node.client.onerror = function (err) {
+                    node.status({ fill: "red", shape: "dot", text: `Error: ${err.message}` });
                 }
             }
-            
+
             // Handle ALL events.
-            node.client.onAnyMessage = function(eventType, event) {
+            node.client.onAnyMessage = function (eventType, event) {
                 handleEvent(event);
             }
         }
-        
-        node.on("input", function(msg) {   
-            node.prevMsg = msg;
-            handleMsg(msg);
+
+        node.on("input", function (msg) {
+            let _msg = RED.util.cloneMessage(msg);
```

Comment:
Could you add a comment "why" it is required to clone the message?  Normally I never clone messages, because if they contain large data (e.g. an image) this statement completely ruins the performance of my Raspberry.

Reply:
Hi Bart
Me also, normally, don't clone, but in this case it helped me sorting out the memory leaking issue. I think something below the pipeline is retaining such values, avoiding the GC to collect the unref variables.
I won't touch the eventsource code, so the simplest way has been to clone the input msg.
The input msg is so little, that cloning it don't affect the performances, thus letting the GC to free such **_msg** once the "input" event has been elaborated.